addons/nuxeo-template-rendering/nuxeo-template-rendering-core/src/main/java/org/nuxeo/template/processors/AbstractBindingResolver.java

```diff
@@ -70,106 +78,179 @@ protected DocumentObjectWrapper getWrapper() {
-                                    context.put(param.getName(), loopVal);
-                                } else {
-                                    context.put(param.getName(), nuxeoWrapper.wrap(property));
-                                }
-                            }
-                        }
-                    } else {
-                        // no available value, try to find a default one ...
-                        if (property != null) {
-                            Type pType = property.getType();
-                            if (pType.getName().equals(BooleanType.ID)) {
-                                context.put(param.getName(), Boolean.FALSE);
-                            } else if (pType.getName().equals(DateType.ID)) {
-                                context.put(param.getName(), new Date());
-                            } else if (pType.getName().equals(StringType.ID)) {
-                                context.put(param.getName(), "");
-                            } else if (pType.getName().equals(InputType.Content.getValue())) {
-                                context.put(param.getName(), "");
-                            } else {
-                                context.put(param.getName(), "!NOVALUE!");
-                            }
-                            // handle special case for pictures
-                            if (param.getType() == InputType.PictureProperty) {
-                                context.put(param.getName(), handlePictureField(param.getName(), null));
-                            }
-                        } else {
-                            if (param.getType().equals(InputType.PictureProperty)) {
-                                context.put(param.getName(), handlePictureField(param.getName(), null));
-                            }
-                        }
+                Property property = getDocProperty(param, doc);
+                if (param.isAutoLoop())
+                    return extractAutoLoop(param, property);
+                else {
+                    try {
+                        return nuxeoWrapper.wrap(property);
+                    } catch (TemplateModelException e) {
+                        throw new ValueNotFound(e);
                     }
+                }
+        }
+    }
+
+    protected void addDefaultMimetypeIfRequired(Blob blob) {
+        if (blob.getMimeType() == null || "".equals(blob.getMimeType().trim())) {
+            blob.setMimeType("image/jpeg");
+        }
+    }
+
+    protected Object extractAutoLoop(TemplateInput param, Property property) {
+        // should do the same on all children properties ?
+        return handleLoop(param.getName(), property);
+    }
+
+    protected Object extractBlobContent(DocumentModel doc, TemplateInput param) throws ValueNotFound {
+        Object propValue = getDocPropertyValue(doc, param.getSource());
+        if (propValue != null && propValue instanceof Blob) {
```

Comment:
`propValue != null` is redundant because `propValue instanceof Blob` already checks for it.